drivers/i2c/i2c_mcux_flexcomm.c

```diff
@@ -208,6 +225,9 @@ static const struct i2c_driver_api mcux_flexcomm_driver_api = {
 	static void mcux_flexcomm_config_func_##id(const struct device *dev); \
 	static const struct mcux_flexcomm_config mcux_flexcomm_config_##id = {	\
 		.base = (I2C_Type *) DT_INST_REG_ADDR(id),		\
+		.clock_name = DT_INST_CLOCKS_LABEL(id),		\
+		.clock_subsys =				\
+		(clock_control_subsys_t)DT_INST_CLOCKS_CELL(id, name),\
```

Comment:
If we encode `ahb_offset`, and `bit` instead of `name` in the DTS this could look something like:

`.clock_subsys = (clock_control_subsys_t)(CLK_GATE_DEFINE(DT_INST_CLOCKS_CELL(id, ahb_offset), DT_INST_CLOCKS_CELL(id, bit))`

Reply:
It's not the clock gate we need here